node/src/components/storage.rs

```diff
@@ -607,8 +601,15 @@ impl Storage {
             NetRequest::BlockAndMetadataByHeight(ref serialized_id) => {
                 let item_id = decode_item_id::<BlockWithMetadata>(serialized_id)?;
                 let opt_item = self
-                    .read_block_and_sufficient_finality_signatures_by_height(item_id)
-                    .map_err(FatalStorageError::from)?;
+                    .read_block_and_metadata_by_height(item_id)
+                    .map_err(FatalStorageError::from)?
+                    .and_then(|block_with_metadata| {
+                        if block_with_metadata.finality_signatures.proofs.is_empty() {
+                            None
+                        } else {
+                            Some(block_with_metadata)
+                        }
```

Comment:
Maybe simpler? (Also below.)

```suggestion
                    .filter(|block_with_metadata| {
                        !block_with_metadata.finality_signatures.proofs.is_empty()
                    })
```

Reply:
Neither `and_then` nor `filter` is required as you noticed below - these have been removed now.